src/nspawn/nspawn.c

```diff
@@ -4436,7 +4486,7 @@ static int outer_child(
                                 return log_error_errno(r, "Failed to join network namespace: %m");
                 }
 
-                if (!arg_privileged) {
+                if (arg_userns_mode == USER_NAMESPACE_MANAGED) {
                         /* In unprivileged operation, sysfs + procfs are special, we'll have to mount them
```

Comment:
Please update the comment. `unprivileged operation` -> `managed namespace operation`